Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -1264,7 +1267,8 @@ ErrorOr<NonnullRefPtr<Custody>> VirtualFileSystem::resolve_path_without_veil(Cre
 
             TRY(validate_path_against_process_veil(*custody, options));
 
-            auto symlink_target = TRY(child_inode->resolve_as_link(credentials, parent, out_parent, options, symlink_recursion_level + 1));
+            CustodyBase base(parent);
```

Comment:
This should work implicitly using lifetime extension (i.e. just keep the `parent` in the `resolve_as_link` line), right?

Same for most/all of the other new occurrences of `CustodyBase(NNRP<Custody>)`.